build_and_push.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -eE
+
+COLOR_RED='\033[0;31m'
+trap 'echo -e "${COLOR_RED}Failed building/pushing container images"' ERR
+
+function usage() {
+    echo "Usage: $0 [-r <registry remote>] image|manifest"
+    exit 0
+}
+
+function echo_run() {
+    echo "$@"
+    "$@"
+}
+
+# default REMOTE
+REMOTE="docker.io/finkandreas"
+
+while getopts "hr:" Option
+do
+  case $Option in
+    r     ) REMOTE=${OPTARG};;
+    h     ) usage;;
+    *     ) echo "Unimplemented option chosen.";;   # DEFAULT
+  esac
+done
+
+shift $(( OPTIND - 1 ))
+
+if [[ "$1" != "image" && "$1" != "manifest" ]] ; then
+    usage
+fi
+
+
+function build_image() {
+    DOCKERFILE="$1" ; shift
+    DOCKERTAG="$1" ; shift
+    echo_run podman build --format docker --pull -f "$DOCKERFILE" -t "$DOCKERTAG" $@ .
+    echo_run podman push "$DOCKERTAG"
+}
+
+function build_manifest() {
+    for tag in "$@" ; do
+        echo_run podman rmi $tag || true
+        echo_run podman manifest rm $tag || true
+        echo_run podman manifest create $tag
+        for arch in x86_64 aarch64 ; do
+            echo_run podman manifest add $tag docker://$tag-$arch
+        done
+        echo_run podman manifest push --all $tag docker://$tag
+    done
+}
+
+
+for spackver in "v0.20.3" "v0.21.0" ; do
+    for baseimg in docker.io/ubuntu:22.04 ; do
+        SPACK_DOCKER_TAG=$(echo $spackver | sed -e 's/^v//')
+        OS_DOCKER_TAG=$(basename "$baseimg" | sed -e 's/://')
+        DOCKER_TAG=${REMOTE}/spack:${SPACK_DOCKER_TAG}-${OS_DOCKER_TAG}
+        BASE_TAG_NAME=${REMOTE}/spack:base-${OS_DOCKER_TAG}
+        if [[ "$1" == "image" ]] ; then
```

Comment:
What about something like this?
```suggestion
        if [[ "$COMMAND" == "image" ]] ; then
```
And somewhere above
```
COMMAND=$1
```